py/ai/h2o/sparkling/ml/algos/H2OAutoML.py

```diff
@@ -16,52 +16,26 @@
-                 weightCol=None, ignoredCols=[], includeAlgos=None, excludeAlgos=None, projectName=None, maxRuntimeSecs=3600.0, stoppingRounds=3,
-                 stoppingTolerance=0.001, stoppingMetric="AUTO", nfolds=5, convertUnknownCategoricalLevelsToNa=True, seed=-1,
-                 sortMetric="AUTO", balanceClasses=False, classSamplingFactors=None, maxAfterBalanceSize=5.0,
-                 keepCrossValidationPredictions=True, keepCrossValidationModels=True, maxModels=0,
-                 predictionCol="prediction", detailedPredictionCol="detailed_prediction", withDetailedPredictionCol=False,
-                 convertInvalidNumbersToNa=False, **deprecatedArgs):
+    def __init__(self, **kwargs):
         Initializer.load_sparkling_jar()
         super(H2OAutoML, self).__init__()
         self._java_obj = self._new_java_obj("ai.h2o.sparkling.ml.algos.H2OAutoML", self.uid)
-
-        self._setDefault(featuresCols=[], labelCol="label", allStringColumnsToCategorical=True, columnsToCategorical=[], splitRatio=1.0, foldCol=None,
-                         weightCol=None, ignoredCols=[], includeAlgos=None, excludeAlgos=None, projectName=None, maxRuntimeSecs=3600.0, stoppingRounds=3,
-                         stoppingTolerance=0.001, stoppingMetric="AUTO", nfolds=5,
-                         convertUnknownCategoricalLevelsToNa=True, seed=-1, sortMetric="AUTO", balanceClasses=False,
-                         classSamplingFactors=None, maxAfterBalanceSize=5.0, keepCrossValidationPredictions=True,
-                         keepCrossValidationModels=True, maxModels=0, predictionCol="prediction", detailedPredictionCol="detailed_prediction",
-                         withDetailedPredictionCol=False, convertInvalidNumbersToNa=False)
-        kwargs = get_input_kwargs(self)
-
-        self.setParams(**kwargs)
-
-    @keyword_only
-    def setParams(self, featuresCols=[], labelCol="label", allStringColumnsToCategorical=True, columnsToCategorical=[], splitRatio=1.0, foldCol=None,
-                  weightCol=None, ignoredCols=[], includeAlgos=None, excludeAlgos=None, projectName=None, maxRuntimeSecs=3600.0, stoppingRounds=3,
-                  stoppingTolerance=0.001, stoppingMetric="AUTO", nfolds=5, convertUnknownCategoricalLevelsToNa=True, seed=-1,
-                  sortMetric="AUTO", balanceClasses=False, classSamplingFactors=None, maxAfterBalanceSize=5.0, keepCrossValidationPredictions=True,
-                  keepCrossValidationModels=True, maxModels=0, predictionCol="prediction", detailedPredictionCol="detailed_prediction",
-                  withDetailedPredictionCol=False, convertInvalidNumbersToNa=False, **deprecatedArgs):
-
-        kwargs = get_input_kwargs(self)
-
-        return self._set(**kwargs)
-
-    def _create_model(self, java_model):
-        return H2OMOJOModel(java_model)
+        self._setDefaultValuesFromJava()
+        self._set(**kwargs)
+
+    def leaderboard(self):
```

Comment:
Moved back as I don't think there is special need to generate the Algos now as the code is really short